server/handler/base.go

```diff
@@ -391,6 +401,31 @@ func selectionToReviewersRequest(s reviewer.Selection) github.ReviewersRequest {
 	return req
 }
 
+func (b *Base) dismissStaleReviewsForResult(ctx context.Context, prctx pull.Context, result common.Result) error {
+	if len(result.DiscardedReviews) == 0 {
+		return nil
+	}
+
+	for _, d := range result.DiscardedReviews {
+		if d.State != pull.ReviewApproved {
+			continue
+		}
+
+		var reasons []string
+		for _, reason := range d.Reason {
+			reasons = append(reasons, string(reason))
+		}
+		because := strings.Join(reasons, " and ")
+		message := fmt.Sprintf("%s was dismissed by policy-bot because the approval was %s", result.Name, because)
```

Comment:
I'm wondering if it's worth simplifying this. Having a reason is helpful, but if a review is dismissed for multiple reasons, I'm not sure it adds much to know all of them instead of the first. What do you think?

Also, once we're accounting for multiple rules dismissing the same review, the `result.Name` part will become kind of arbitrary and I don't think will add much. Rule names can also be quite long, which might hit character limits in the GitHub API.

Reply:
yea, i think you're right, putting user input like the rule names in the message is asking for character limit problems, we should probably just keep this simple with just the reason and i think you're right, it doesn't really provide much value to include both.